src/Components/Facility/Consultations/PressureSoreDiagrams.tsx

```diff
@@ -19,49 +20,51 @@ export const PressureSoreDiagrams = (props: any) => {
   const [currentPage, setCurrentPage] = useState(1);
   const [totalCount, _setTotalCount] = useState(0);
 
-  const fetchDailyRounds = useCallback(
-    async (status: statusType) => {
+  useEffect(() => {
+    const fetchDailyRounds = async (
+      currentPage: number,
+      consultationId: string
+    ) => {
       setIsLoading(true);
-      const res = await dispatch(
-        dailyRoundsAnalyse(
-          {
+      const { res, data: dailyRounds } = await request(
+        routes.dailyRoundsAnalyse,
+        {
+          body: {
             page: currentPage,
             fields: ["pressure_sore"],
           },
-          { consultationId }
-        )
+          pathParams: {
+            consultationId,
+          },
+        }
       );
-      if (!status.aborted) {
-        if (res && res.data) {
-          const keys = Object.keys(res.data.results || {}).filter(
-            (key) => res.data.results[key].pressure_sore.length
-          );
-          const data: any = {};
-          keys.forEach((key) => (data[key] = res.data.results[key]));
+      if (res && res.ok && dailyRounds) {
+        const keys = Object.keys(dailyRounds.results || {}).filter(
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
```

Comment:
You are not allowed do ts-ignore, when you are improving type-safety. The `data` has a proper type, so you should use them instead of muting typescript errors.

```suggestion
```